mlrun/model.py

```diff
@@ -1551,12 +1555,20 @@ def _outputs_wait_for_completion(
             )
 
     def _artifact(self, key):
-        """return artifact DataItem by key"""
-        if self.status.artifacts:
-            for a in self.status.artifacts:
-                if a["metadata"]["key"] == key:
-                    return a
-        return None
+        """Return artifact DataItem by key."""
+        if not self.status.artifacts:
+            return None
+
+        # Find the first artifact that matches the given key and tag is not 'latest'
+        # Or None if no such artifact is found
+        return next(
+            (
+                a
+                for a in self.status.artifacts
+                if a["metadata"]["key"] == key and a["metadata"]["tag"] != "latest"
```

Comment:
```suggestion
                artifact
                for artifact in self.status.artifacts
                if artifact["metadata"]["key"] == key and a["metadata"]["tag"] != "latest"
```
[coding convention - 5](https://github.com/mlrun/mlrun/blob/development/CONTRIBUTING.md#python-code-conventions)